src/Equinox.CosmosStore/CosmosStore.fs

```diff
@@ -384,22 +384,18 @@ module Log =
 [<AutoOpen>]
 module private MicrosoftAzureCosmosWrappers =
 
-    type Headers with
-        member headers.GetRequestCharge () =
-            match headers.TryGetValue "x-ms-request-charge" with
-            | true, charge -> float charge
-            | _ -> 0.
-
     type ReadResult<'T> = Found of 'T | NotFound | NotModified
     type Container with
-        member container.TryReadItem(partitionKey : PartitionKey, documentId : string, ?options : ItemRequestOptions): Async<float * ReadResult<'T>> = async {
+        member private container.DeserializeResponseBody<'T>(rm : ResponseMessage) : 'T =
```

Comment:
@ealsur could something like this be exposed on the SDK in order to make the deferred deserialization clean and not reliant on spelunking in the source? (Maybe as an extension method on the `ResponseMessage` either?)

Reply:
while I think the code in https://github.com/jet/equinox/blob/eade429557a857fedbaa0227ec28358f90a3bc50/src/Equinox.CosmosStore/CosmosStore.fs#L391-L395 is pretty clear for my context, it's not obvious and the xmldoc does not hint at it being possible 
(also, It's very possible my impl is not optimal, in which case I'd like to be using an SDK method that will be)

Reply:
Sorry for the late response, missed this notification. Stream APIs are by nature unhooked from serialization, so the user can choose whatever serialization engine they want (or simply return the Stream to a caller if you are working in a middleware).

We don't expect users to consume the Stream APIs and also use our own Serializer, in that case, why are you using the Stream API? (sorry, my F# is non-existent, so I might be missing something).

Reply:
This was via https://twitter.com/OliTomli/status/1466055639906590723
Recall non-200s throw on the non-stream API - the code above is derived from V2 SDK code and eventually could/will do its own deserialization (using STJ) but my real broad point is that adding a helper like this or maybe an overload with a callbck would allow one to mix and match the Stream and non-Stream variants when you discover you want to filter out or provide alternate processing for a given status